src/tic_tac_toe/board.clj

```diff
@@ -1,11 +1,84 @@
 (ns tic-tac-toe.board)
 
+(def empty-mark "-")
+
+(defn- empty-cell? [cell]
+  (= empty-mark cell))
+
+(defn- get-cells [board positions]
+  (mapv board positions))
+
+(defn- contains-blank-cell? [board]
+  (if (some #(= empty-mark %) board) true false))
+
+(defn- cells-the-same? [row board]
+  (if (contains-blank-cell? (row board)) 
+    false
+    (apply = (row board))))
+
+(defn- top-row-the-winner? [board]
+  (cells-the-same? #(get-cells % [0 1 2])board))
+
+(defn- middle-row-the-winner? [board]
+  (cells-the-same? #(get-cells % [3 4 5])board))
+
+(defn- bottom-row-the-winner? [board]
+  (cells-the-same? #(get-cells % [6 7 8])board))
+
+(defn- left-column-winner? [board]
+  (cells-the-same? #(get-cells % [0 3 6])board))
+
+(defn- middle-column-winner? [board]
+  (cells-the-same? #(get-cells % [1 4 7])board)) 
+
+(defn- right-column-winner? [board]
+  (cells-the-same? #(get-cells % [2 5 8])board)) 
+
+(defn- diagonal-top-left-winner? [board]
+  (cells-the-same? #(get-cells % [0 4 8])board))
+
+(defn- diagonal-top-right-winner? [board]
+  (cells-the-same? #(get-cells % [2 4 6])board))
+
+(defn- board-empty? [board]
+  (every? #{empty-mark} board))
+
```

Comment:
what about something like? (not tested)

`(defn- winner? [board]  (some (partial foo board) [[1 2 3] [2 5 8] [0 4 8] [2 4 6]])) `
`(defn- foo[board line] (cells-the-same? #(get-cells % line)board)))`